cirq/linalg/operator_spaces.py

```diff
@@ -0,0 +1,69 @@
+# Copyright 2018 The Cirq Developers
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Utilities for manipulating linear operators as elements of vector space."""
+
+from typing import Dict, Tuple
+
+import numpy as np
+
+PAULI_BASIS = {
+    'I': np.eye(2),
+    'X': np.array([[0., 1.], [1., 0.]]),
+    'Y': np.array([[0., -1j], [1j, 0.]]),
+    'Z': np.diag([1., -1]),
+}
+
+
+def kron_bases(basis1: Dict[str, np.ndarray],
```

Comment:
I think this should have the same signature as `itertools.product`, e.g. you can specify an arbitrary number of bases and also there's a `repeat` parameter.

Reply:
Good idea! Done and added tests to cover repeats.